include/bringauto/internal_server/InternalServer.hpp

```diff
@@ -81,10 +81,11 @@ class InternalServer {
 	 * @brief Processes buffer data and once message is complete calls handleMessage(...)
 	 * @param connection conneection with context holding received and processed data
 	 * @param bytesTransferred  size of received data
+	 * @param bufferOffset offset of buffer where data starts
 	 * @return true if data and whole message is correct in context to fleet protocol
 	 */
 	bool processBufferData(const std::shared_ptr<structures::Connection> &connection,
-						   std::size_t bytesTransferred);
+						   std::size_t bytesTransferred, std::size_t bufferOffset = 0);
```

Comment:
hmmm... the argument bufferOffset does not make sense from sema=ntics point of view. Which buffer offset? It should be part of the connContext but not as an argument of the func....

Reply:
decided in conversation that it makes sense, added checks for overflows